server/src/controllers/authentication_controller.ts

```diff
@@ -1,44 +1,95 @@
-        const schema: Joi.ObjectSchema = Joi.object({
-            email: Joi.string().email(),
-            // Password requires upper case and lower case letter as well as number and special character.
-            password: Joi.string()
-                .pattern(new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z])(?=.*[!@#$%^&*()?+=_]).{8,30}$"))
-                .required(),
-            first_name: Joi.string().pattern(new RegExp("^[A-Za-z]+")).min(2).max(25).required(),
-            last_name: Joi.string().pattern(new RegExp("^[A-Za-z]+")).min(2).max(25).required(),
-            date_of_birth: Joi.date().required(),
-            created_on: Joi.date().timestamp(),
-        });
-
-        const { error } = await schema.validateAsync(req.body);
-
-        if (!req.is("application/json") || error) {
-            // indicate error
-            return res.send(400, error || "Data is not in json format.");
+    @Post("/sign_up")
+    private async signUp(req: Request, res: Response): Promise<void> {
+        const { value, error } = signUpSchema.validate(req.body);
+
+        if (error) {
+            res.json(400, error);
+            return;
         }
 
-        const e = await this._authenticationService.createUser(req.body);
-        if (!e) {
-            return res.send(200, "New user has been created.");
+        const userData: RequestUser = {
+            firstName: value.firstName,
+            lastName: value.lastName,
+            phoneNumber: value.phoneNumber,
+            gender: value.gender,
+            dateOfBirth: value.dateOfBirth,
+            email: value.email,
+            password: value.password,
+        };
+        const addressData: RequestAddress = {
+            streetAddress: value.streetAddress,
+            streetAddressLineTwo: value.streetAddressLineTwo,
+            city: value.city,
+            postalCode: value.postalCode,
+            province: value.province,
+            country: "canada",
+        };
+
+        await this.authenticationService
+            .signUp(userData, addressData)
+            .then((token) => res.json(201, { token }))
+            .catch((error) => res.json(500, { error: error.toString() }));
+    }
+
+    @Post("/sign_in")
+    private async signIn(req: Request, res: Response): Promise<void> {
+        const { value, error } = signInSchema.validate(req.body);
+
+        if (error) {
+            res.json(400, error);
+            return;
         }
-        return res.send(500, "New user was not created.");
+
+        await this.authenticationService
+            .signIn(value.email, value.password)
+            .then((token) => res.json(200, { token }))
+            .catch((error) => res.json(500, { error: error.toString() }));
     }
 }
+
+// Password requires at least 1 upper case, 1 lower case letter, 1 number, 1 special character, and
+// has a minimum length of 8 characters
+const passwordSchema = Joi.string()
```

Comment:
Maybe put this in either the same service class as mentioned above or in a validation service class for better code separation.